config/config.go

```diff
@@ -107,6 +107,63 @@ func NewConfigData(adapterName string, data []byte) (Configer, error) {
 	return adapter.ParseData(data)
 }
 
+// ChooseRealValueForMap convert all string value with environment variable.
+func ChooseRealValueForMap(m map[string]interface{}) map[string]interface{} {
+	for k, v := range m {
+		switch value := v.(type) {
+		case string:
+			m[k] = ChooseRealValue(value)
+		case map[string]interface{}:
+			m[k] = ChooseRealValueForMap(value)
+		case map[string]string:
+			for k2, v2 := range value {
+				value[k2] = ChooseRealValue(v2)
+			}
+			m[k] = value
+		}
+	}
+	return m
+}
+
+// ChooseRealValue returns value of convert with environment variable.
+//
+// Return environment variable if value start with "$$".
+// Return default value if environment variable is empty or not exist.
+//
+// It accept value formats "$$env" , "$$env||" , "$$env||defaultValue" , "defaultvalue".
+// Examples:
+//	v1 := config.ChooseRealValue("$$GOPATH")			// return the GOPATH environment variable.
+//	v2 := config.ChooseRealValue("$$GOAsta||/usr/local/go/")	// return the default value "/usr/local/go/".
+//	v3 := config.ChooseRealValue("Astaxie")				// return the value "Astaxie".
+func ChooseRealValue(value string) (realValue string) {
```

Comment:
I think we need to discuss this name. It not make sense for me. @pjoe do you have suggestion?